ui-frontend/packages/catalog-ui-search/src/main/webapp/js/model/QueryResult.js

```diff
@@ -198,38 +198,57 @@ module.exports = Backbone.AssociatedModel.extend({
         url: './internal/cql',
         data: JSON.stringify(req),
         contentType: 'application/json',
-      }).then(this.parseRefresh.bind(this), this.handleRefreshError.bind(this))
+      }).then(
+        this.parseRefresh(metacardProperties).bind(this),
+        this.handleRefreshError.bind(this)
+      )
     }, 1000)
   },
   handleRefreshError() {
     //do nothing for now, should we announce this?
   },
-  parseRefresh(response) {
-    const queryId = this.get('metacard').get('queryId')
-    const color = this.get('metacard').get('color')
-    _.forEach(response.results, (result) => {
-      delete result.relevance
-      result.propertyTypes =
-        response.types[result.metacard.properties['metacard-type']]
-      result.metacardType = result.metacard.properties['metacard-type']
-      result.metacard.id = result.metacard.properties.id
-      result.id = result.metacard.id + result.metacard.properties['source-id']
-      result.metacard.queryId = queryId
-      result.metacard.color = color
-      humanizeResourceSize(result)
-      result.actions.forEach((action) => (action.queryId = queryId))
-      const thumbnailAction = _.findWhere(result.actions, {
-        id: 'catalog.data.metacard.thumbnail',
+  parseRefresh(metacardProperties) {
+    return function (response) {
+      const queryId = this.get('metacard').get('queryId')
+      const color = this.get('metacard').get('color')
+      _.forEach(response.results, (result) => {
+        delete result.relevance
+        result.propertyTypes =
+          response.types[result.metacard.properties['metacard-type']]
+        result.metacardType = result.metacard.properties['metacard-type']
+        result.metacard.id = result.metacard.properties.id
+        result.id = result.metacard.id + result.metacard.properties['source-id']
+        result.metacard.queryId = queryId
+        result.metacard.color = color
+        humanizeResourceSize(result)
+        result.actions.forEach((action) => (action.queryId = queryId))
+        const thumbnailAction = _.findWhere(result.actions, {
+          id: 'catalog.data.metacard.thumbnail',
+        })
+        if (result.hasThumbnail && thumbnailAction) {
+          result.metacard.properties.thumbnail = generateThumbnailUrl(
+            thumbnailAction.url
+          )
+        } else {
+          result.metacard.properties.thumbnail = undefined
+        }
       })
-      if (result.hasThumbnail && thumbnailAction) {
-        result.metacard.properties.thumbnail = generateThumbnailUrl(
-          thumbnailAction.url
-        )
-      } else {
-        result.metacard.properties.thumbnail = undefined
+      const updatedResult = response.results[0]
+      let clearedAttributes = []
+      if (metacardProperties !== undefined) {
+        updatedResult.metacard.properties = metacardProperties
```

Comment:
Do we need to do the fetch if we have the properties on hand?  

Trying to understand if we can short circuit the refreshData function to skip that network call when properties are passed in.

Reply:
I think so, since `response.results[0]` contains things other than the metacard properties 
![image](https://user-images.githubusercontent.com/4467636/114776537-6e9cfe80-9d27-11eb-9bcc-2cd5a521f83e.png)

Reply:
Those are out of date though right, which is the reason we need those properties?